inc/customizer.php

```diff
@@ -3,218 +3,42 @@
-					'css'     => array(
-						'.secondary_text_color, .site-info-text' => array(
-							'color' => '%1$s',
-						),
-					),
-				),
-			)
-		);
+		foreach( glob( dirname( __FILE__ ) . '/customizer/*.php' ) as $filename ) {
 
-		/**
-		 * Default color scheme.
-		 *
-		 * The `default` color scheme is required and not filterable.
-		 * If you want to customize values in this scheme, do so via
-		 * a `primer_colors` filter in your Child Theme.
-		 *
-		 * @var array
-		 */
-		$default_scheme = array(
-			'default' => array(
-				'label'  => __( 'Default', 'primer' ),
-				'colors' => wp_list_pluck( self::$colors, 'default', 'name' ),
-			),
-		);
+			require_once $filename;
 
-		/**
-		 * Filter the available color schemes.
-		 *
-		 * @since 1.0.0
-		 *
-		 * @var array
-		 */
-		$custom_schemes = (array) apply_filters( 'primer_color_schemes',
-			array(
-				'dark' => array(
-					'label'  => __( 'Dark', 'primer' ),
-					'colors' => array(
-						'header_textcolor'      => '#1a1a1a',
-						'background_color'      => '#262626',
-						'menu_background_color' => '#589ef2',
-						'tagline_text_color'    => '#1a1a1a',
-						'link_color'            => '#589ef2',
-						'main_text_color'       => '#e5e5e5',
-						'secondary_text_color'  => '#c1c1c1',
-					),
-				),
-				'muted' => array(
-					'label'  => __( 'Muted', 'primer' ),
-					'colors' => array(
-						'header_textcolor'      => '#5a6175',
-						'background_color'      => '#d5d6e0',
-						'menu_background_color' => '#5a6175',
-						'tagline_text_color'    => '#888c99',
-						'link_color'            => '#3e4c75',
-						'main_text_color'       => '#4f5875',
-						'secondary_text_color'  => '#888c99',
-					),
-				),
-				'red' => array(
-					'label'  => __( 'Red', 'primer' ),
-					'colors' => array(
-						'header_textcolor'      => '#222222',
-						'background_color'      => '#ffffff',
-						'menu_background_color' => '#640c1f',
-						'tagline_text_color'    => '#999999',
-						'link_color'            => '#640c1f',
-						'main_text_color'       => '#402b30',
-						'secondary_text_color'  => '#222222',
-					),
-				),
-			)
-		);
+			$basename  = basename( $filename, '.php' );
+			$classname = 'Primer_Customizer_' . ucfirst( $basename );
+			$classname = apply_filters( 'primer_customizer_component_' . $basename, $classname );
 
-		self::$color_schemes = $default_scheme + $custom_schemes;
+			if ( ! class_exists( $classname ) ) {
 
-		/**
-		 * Filter the array of available fonts.
-		 *
-		 * @since 1.0.0
-		 *
-		 * @var array
-		 */
-		self::$fonts = (array) apply_filters( 'primer_fonts',
-			array(
-				'Open Sans',
-				'Source Sans Pro',
-				'Roboto',
-				'Lato',
-				'Montserrat',
-				'Raleway',
-				'PT Sans',
-				'Noto Sans',
-				'Muli',
-				'Oxygen',
-				'Source Serif Pro',
-				'PT Serif',
-			)
-		);
+				trigger_error( $classname . " doesn't exist.", E_ERROR );
+
+				return;
+
+			}
+
+			$this->$basename = new $classname;
```

Comment:
@jonathanbardo What if we just require the file and it instantiates itself, like this `Primer_Customizer` class does? Also I'm unsure what `$this->$basename` is?